google/cloud/spanner/database_admin_connection_test.cc

```diff
@@ -912,6 +912,10 @@ TEST(DatabaseAdminConnectionTest, CreateBackupSuccess) {
 /// @test Verify that using an encryption key works.
 TEST(DatabaseAdminClientTest, CreateBackupWithEncryption) {
   auto mock = std::make_shared<MockDatabaseAdminStub>();
+  // Suppress a false leak.
+  // TODO(#4038): After we fix the issue #4038, we won't need to use
+  // `AllowLeak()` any more.
+  Mock::AllowLeak(mock.get());
```

Comment:
It occurs to me (too late I know) that we may have the same problem with other mocks in this test ... I am not sure why this mock would be any more leaky than the others... 🤷

Reply:
My rationale (without actually checking) was that only certain test cases hit the detached-thread-creating interfaces, and therefore to play a wait-and-see game for now.